src/vm/assemblynative.cpp

```diff
@@ -1331,3 +1367,26 @@ BOOL QCALLTYPE AssemblyNative::InternalTryGetRawMetadata(
 
     return metadata != nullptr;
 }
+
+INT32 QCALLTYPE AssemblyNative::ExecuteMainMethod(
+    QCall::AssemblyHandle pAssembly,
+    QCall::ObjectHandleOnStack args)
+{
+    QCALL_CONTRACT;
+
+    INT32 exitCode;
+
+    BEGIN_QCALL;
+
+    _ASSERTE(pAssembly != nullptr);
+
+    Assembly* assembly = pAssembly->GetAssembly();
+
+    PTRARRAYREF* argsArray = (PTRARRAYREF*)args.m_ppObject;
```

Comment:
Is this okay to be null, do we need a check or does `ExecuteMainMethod` handle that?

Reply:
I've added some null checking in the managed entrypoint to this call.